storage/innobase/include/buf0flu.ic

```diff
@@ -89,6 +89,9 @@ buf_flush_note_modification(
 	ut_ad(block->page.flush_observer == NULL
 	      || block->page.flush_observer == observer);
 	block->page.flush_observer = observer;
+        if (observer != NULL) {
```

Comment:
I re-read our Friday discussion, and reviewed FlushObserver usage in the code, but I am still unable to convince myself why is this correct - it seems to depend on the page write pattern with FlushObserver active, that there is roughly one mtr per page dirtied. Is this the case with this and other uses?

Reply:
Right, PageBulk in btr0bulk.h has one mtr and pageBulk is per page. The reason I was confident that many duplicates wont exist in this pattern is because a new page is not allocated until the last one on each level is full. 

Currently FlushObserver is not used anywhere else except ALTER ADD INDEX.

But I have another idea (this is easier to understand, general purpose and should satisfy the requirements)

1. Create observer_lsn. The LSN at which flushobserver is created. (Assign log_sys->lsn to observer_lsn)
2. For mtrs with observers, add the page to tracking (increment number of pages) if
    if (oldest_modification == 0 /* new blocks that are added to flush list */
        || oldest_modification < observer->m_observe_lsn /* existing blocks that are dirtied before flush observer */
        observer->inc_estimate() 
   }

This becomes very accurate for an estimate ;) but I am OK with implementing this :)

Reply:
I like this suggestion

Reply:
One small correction, observer->lsn() should be compared with newest modification not oldest modification.